resharper/resharper-unity/src/Unity.Rider/UnityEditorIntegration/RiderPackageUpdateAvailabilityChecker.cs

```diff
@@ -0,0 +1,158 @@
+                BindToInstallationSettingChange(lt, leastRiderPackageVersion);
+                BindToProtocol(lt);
+            });
+        }
+
+        private void BindToProtocol(Lifetime lt)
+        {
+            myBackendUnityHost.BackendUnityModel.ViewNotNull(lt, (l, model) =>
+            {
+                model.RiderPackagePotentialUpdateVersion.Advise(l, result =>
+                {
+                    if (!string.IsNullOrEmpty(result) && Version.TryParse(result, out var resultVersion))
+                    {
+                        ShowNotificationIfNeeded(l, resultVersion);
+                    }
+                });
+            });
+        }
+
+        private void BindToInstallationSettingChange(Lifetime lifetime, Version version)
+        {
+            var entry = myBoundSettingsStore.Schema.GetScalarEntry((UnitySettings s) =>
+                s.AllowRiderUpdateNotifications);
+            myBoundSettingsStore.GetValueProperty<bool>(lifetime, entry, null).Change.Advise_NoAcknowledgement(lifetime,
+                args =>
+                {
+                    if (!args.GetNewOrNull()) return;
+                    ShowNotificationIfNeeded(lifetime, version);
+                });
+        }
+
+        private void ShowNotificationIfNeeded(Lifetime lifetime, Version expectedVersion)
+        {
+            if (!myBoundSettingsStore.GetValue((UnitySettings s) => s.AllowRiderUpdateNotifications))
+                return;
+
+            myPackageManager.IsInitialUpdateFinished.WhenTrue(lifetime, lt =>
+            {
+                myUnityVersion.ActualVersionForSolution.AdviseNotNull(lt, version =>
+                {
+                    if (myNotificationShown.Contains(mySolution.SolutionFilePath)) return;
+
+                    // Version before 2019.2 doesn't have Rider package
+                    // 2019.2.0 - 2019.2.5 : version 1.2.1 is the last one
+                    // 2019.2.6 - present : see: leastRiderPackageVersion
+                    if (version < new Version(2019, 2, 6)) return;
+
+                    var package = myPackageManager.GetPackageById(packageId);
+
+                    if (package == null)
+                    {
+                        myNotificationShown.Add(mySolution.SolutionFilePath);
+                        myLogger.Info($"{packageId} is missing.");
+                        var notification = RiderNotification.Create(NotificationSeverity.WARNING,
+                            "JetBrains Rider package in Unity is missing.",
+                            "Make sure JetBrains Rider package is installed in Unity Package Manager."
+                        );
+                        myShellLocks.ExecuteOrQueueEx(lt,
+                            "RiderPackageUpdateAvailabilityChecker.ShowNotificationIfNeeded",
+                            () =>
+                            {
+                                myNotificationPopupHost.ShowNotification(lt, notification);
+                            });
```

Comment:
I think this can be simplified to just `UserNotifications.CreateNotiification`. AFAICT, this will both create and show the notification with one API.